batavia/builtins/bool.js

```diff
@@ -17,7 +18,16 @@ function bool(args, kwargs) {
     if (args[0] === null) {
         return new types.NoneType.__bool__()
     } else if (args[0].__bool__) {
-        return args[0].__bool__()
+        // args[0].__bool__, if it exists, is a batavia.types.Function.js,
+        // *not* a native Javascript function. Therefore we can't call it in
+        // the seemingly obvious way, with __bool__().
+        return callables.call_method(args[0], '__bool__', [])
+    // Python bool() checks for __bool__ and then, if __bool__ is not defined,
+    // for __len__. See https://docs.python.org/3.4/library/stdtypes.html#truth.
+    } else if (args[0].__len__) {
+        // valueOf() can return a string even when __len__ was set to an
+        // integer (as in builtins.test_bool.BoolTests.test_len_only).
+        return !!parseInt(callables.call_method(args[0], '__len__', []).valueOf())
```

Comment:
I thought about that and decided I was unsure. Mostly because I realized I don't actually trust people to have implemented `__len__` in such a way that it returns something integer-like, so if someone's written a __len__ that returns, say, "seventeen" or ⛔️ , I'd rather have something that I'm pretty sure will fail dramatically.

I didn't have the time at the sprint to think through all the angles on that question, so I'm open to other arguments :)